model/task/generate_tasks_estimations.go

```diff
@@ -0,0 +1,72 @@
+package task
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/utility"
+	adb "github.com/mongodb/anser/db"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	numRevisionsToSearch = 10
+)
+
+func (t *Task) setGenerateTasksEstimations() error {
+	if !t.GenerateTask || (t.EstimatedNumGeneratedTasks != nil && t.EstimatedNumActivatedGeneratedTasks != nil) {
+		return nil
+	}
+
+	query := db.Query(
+		ByPreviousCommit(
+			t.BuildVariant,
+			t.DisplayName,
+			t.Project,
+			evergreen.RepotrackerVersionRequester,
+			t.RevisionOrderNumber,
+		),
+	).Sort([]string{"-" + RevisionOrderNumberKey}).Limit(numRevisionsToSearch)
+
+	tasks := []Task{}
+	err := db.FindAllQ(Collection, query, &tasks)
+	if adb.ResultsNotFound(err) {
+		return errors.Errorf("Tasks '%s' in '%s' not found", t.DisplayName, t.Project)
+	}
+	if err != nil {
+		return errors.Wrapf(err, "finding tasks '%s' in '%s'", t.DisplayName, t.Project)
+	}
+	if len(tasks) == 0 {
+		return errors.Errorf("No tasks found for '%s' in '%s'", t.DisplayName, t.Project)
```

Comment:
It seems like, rather than return an error, it would make sense to have some kind of default (even if it's zero) in the case where a new task is committed to the waterfall, so it won't have an estimated number of tasks.